### PR TITLE
Fix HTTP basic auth (fix #1219)

### LINKED ISSUE #1219: [regression-44.0.1] Basic HTTP auth (username+password) is broken
After last fix to 44.0.1 im getting: 
```
Unauthorized
This server could not verify that you are authorized to access the document requested. Either you supplied the wrong credentials (e.g., bad password), or your browser doesn't understand how to supply the credentials required.

Additionally, a 401 Unauthorized error was encountered while trying to use an ErrorDocument to handle the request.
```

**On Mac and windows**

 I am using among others that might be related:  
`--basic-auth-password  --basic-auth-username --insecure --ignore-certificate  `

On 43.1.3 and 44.0.0 this is not a problem.

### PR DESCRIPTION
**Pull request by TheCleric**
Should resolve #1219 as well adds a manual test for basic auth as suggested.

### CODE REVIEW

**Comment by TheCleric** [comment]
Don't merge this just yet @ronjouch . This fixes the basic-auth params, but the login prompt (when not supplying) is acting funky.

**Comment by TheCleric** [comment]
Okay should be good now. Had to implement the suggestion from https://github.com/electron/electron/issues/28017

File: .github/manual-test [hunk 1]
```diff
@@ -48,14 +49,102 @@ fi
...
+printf "\n***** Running app *****\n"
+if [ "$(uname -s)" = "Darwin" ]; then
+  open -a "$tmp_dir/app-darwin-x64/app.app"
+else
+  "$tmp_dir/app-linux-x64/app"
+fi
+
+printf "\nDid everything work as expected? [yN] "
+read -r response
+
```
Comment (ronjouch) [inline .github/manual-test:138]: Is this third test really useful? Remember we're in manual land here, we don't want this script to become giant

File: app/src/components/loginWindow.ts [hunk 2]
```diff
@@ -18,13 +18,15 @@ export async function createLoginWindow(
     resizable: false,
     webPreferences: {
       nodeIntegration: true, // TODO work around this; insecure
+      contextIsolation: false,
```
Comment (ronjouch) [inline app/src/components/loginWindow.ts:21]: In-code comment, pls

File: app/src/components/loginWindow.ts [hunk 3]
```diff
@@ -18,13 +18,15 @@ export async function createLoginWindow(
...
     },
   });
   await loginWindow.loadURL(
     `file://${path.join(__dirname, 'static/login.html')}`,
   );
 
-  ipcMain.once('login-message', (event, usernameAndPassword) => {
+  ipcMain.on('login-message', (event, usernameAndPassword) => {
```
Comment (ronjouch) [inline app/src/components/loginWindow.ts:28]: Same here: pls comment and link to https://github.com/electron/electron/issues/28017

File: app/src/main.ts [hunk 4]
```diff
@@ -229,67 +231,60 @@ if (appArgs.widevine) {
...
+      log.error('createLoginWindow ERROR', err),
+    );
+  }
+});
 
 async function onReady(): Promise<void> {
-  const mainWindow = await createMainWindow(appArgs, setDockBadge);
-
-  initAppWindowEvents(mainWindow);
+  mainWindow = await createMainWindow(appArgs, setDockBadge);
```
Comment (ronjouch) [inline app/src/main.ts:287]:
```suggestion
  // Warning: `mainWindow` below is the *global* unique `mainWindow`, created at init time
  mainWindow = await createMainWindow(appArgs, setDockBadge);  
```

File: .github/manual-test [hunk 5]
```diff
@@ -48,14 +49,102 @@ fi
...
+    }
+- Console: no Electron runtime deprecation warnings/error logged
+"
+
+printf "\n***** Running app *****\n"
+if [ "$(uname -s)" = "Darwin" ]; then
+  open -a "$tmp_dir/app-darwin-x64/app.app"
+else
+  "$tmp_dir/app-linux-x64/app"
+fi
```
Comment (ronjouch) [inline .github/manual-test:90]: Pls extract into a lil' `launch_app` helper function defined at the beginning of the script

File: .github/manual-test [hunk 6]
```diff
@@ -48,14 +49,102 @@ fi
...
+  rm -rf "$tmp_dir"
+  printf "\n"
+fi
+
+if [ "$response" != 'y' ]; then
+  echo "Back to fixing"
+  exit 1
+else
+  echo "Yayyyyyyyyyyy"
+fi
```
Comment (ronjouch) [inline .github/manual-test:64]: Pls extract into a reused `ask_for_feedback` helper function

**Review by ronjouch — CHANGES_REQUESTED** [review]
Thx @TheCleric . A few PR nits, and most importantly needs comments about the contextIsolation/ipc thing

File: .github/manual-test (hunk 1 above)
Comment (TheCleric) [reply .github/manual-test:138]: It's as useful as making sure the login dialog works. If that's not a huge worry we can remove.

Comment (ronjouch) [reply .github/manual-test:138]: > It's as useful as making sure the login dialog works. If that's not a huge worry we can remove.

Sry, didn't understand it hit the login dialog, I misread it as testing basically nothing. 👍.

**Comment by TheCleric** [comment]
@ronjouch all nits should be addressed!

**Review by ronjouch — APPROVED** [review]
all-the-things

**Pull request merged**